### DISCUSSION TITLE
Compiler Version Error: solidity 0.8.20 not working for Raffle Project

### DISCUSSION THREAD

**Discussion by simon-masterclass** [Cyfrin/foundry-full-course-cu#2019] [General]
Hello Cyfrin team & Patrick, 
First of all, big thanks to @PatrickAlphaC and the team for putting this AWESOME course together. I've learned more from this course than all the other Solidity bootcamps courses (some expensive ones too) combined!

I'm working my way through **Section 4:  Smart Contract Lottery** of the Foundry Fundamentals course - Lesson 21 specifically.
I'm having trouble compiling with any version of solidity higher than 0.8.19
I went through all the steps of asking Ai (chatGPT), etc and it looks like the issue is in a Chainlink-Brownie-Contract file related to vrf contracts - specifically one of the inherited contracts. 

SubscriptionAPI.sol
@chainlink-brownie-contracts/contracts/src/v0.8/vrf/dev/SubscriptionAPI.sol

It appears that the pragma solidity version is missing a "^" (Carot) symbol, thereby hampering the ability to compile with any version of solidity greater than 0.8.19 - which might be intentional or a bug.

Here's the code:
```Solidity
// SPDX-License-Identifier: MIT
pragma solidity 0.8.19;

import {EnumerableSet} from "../../vendor/openzeppelin-solidity/v4.7.3/contracts/utils/structs/EnumerableSet.sol";
import {LinkTokenInterface} from "../../shared/interfaces/LinkTokenInterface.sol";
import {ConfirmedOwner} from "../../shared/access/ConfirmedOwner.sol";
import {AggregatorV3Interface} from "../../shared/interfaces/AggregatorV3Interface.sol";
import {IERC677Receiver} from "../../shared/interfaces/IERC677Receiver.sol";
import {IVRFSubscriptionV2Plus} from "./interfaces/IVRFSubscriptionV2Plus.sol";

abstract contract SubscriptionAPI is ConfirmedOwner, IERC677Receiver, IVRFSubscriptionV2Plus {
  using EnumerableSet for EnumerableSet.UintSet;
```

Is this a bug or a feature?
Let me know, thanks!

-Simon

**Comment by EngrPips** [comment]
Hello @simon-masterclass, Thanks for the good words. The Cyfrins `Foundry and Solidity course is the best I have seen myself. Honestly, I don't know if this that you pointed out is a bug or a feature, but I changed the pragma to make it work for me, lol.

```js
// SPDX-License-Identifier: MIT
pragma solidity 0.8.23;

import {EnumerableSet} from "../../vendor/openzeppelin-solidity/v4.7.3/contracts/utils/structs/EnumerableSet.sol";
import {LinkTokenInterface} from "../../shared/interfaces/LinkTokenInterface.sol";
import {ConfirmedOwner} from "../../shared/access/ConfirmedOwner.sol";
import {AggregatorV3Interface} from "../../shared/interfaces/AggregatorV3Interface.sol";
import {IERC677Receiver} from "../../shared/interfaces/IERC677Receiver.sol";
import {IVRFSubscriptionV2Plus} from "./interfaces/IVRFSubscriptionV2Plus.sol";
import {console2} from "forge-std/Script.sol";
```
I am looking forward to Patrick's comments.

**Reply by simon-masterclass** [reply]
Yea, and that's not the only chainlink-brownie-contracts file with a missing ^ symbol - there are several others. 

I also noticed that some of the contracts are using an older version of the OpenZeppelin library (probably <5.0) which has a different file structure. 
For example, Pausable.sol used to reside in the folder...
@OpenZeppelin/contracts/security/Pausable.sol 
but now it's been moved to...
@OpenZeppelin/contracts/utils/Pausable.sol
 Which led to some reference errors in some of the VRF related contracts.

The Chainlink-brownie-contracts repo is probably overdue for some maintenance...

**Reply by EngrPips** [reply]
Yeah, There are some changes here and there, and I trust all of those will be accounted for appropriately. Thanks for pointing stuff out, buddy.

**Reply by PatrickAlphaC** [reply]
As of today, the Chainlink repo needs an update :( 

The work around would be to add that little carrot `^` into your `lib` folder manually. And then keep bothering Chainlink Labs (`smartcontractkit` org on GitHub) to fix it.

Or, you can just use `0.8.19`

**Reply by simon-masterclass** [reply]
Ok cool, I've been fixing little reference errors and missing ^ (carots) here and there. It seems like the issue isn't in the Chainlink/contracts repo - only the Chainlink-brown-contracts repo has issues with missing ^
Definitely in need of some updating but certainly not something that can't be overcome with a few minor modifications. 
It doesn't seem like many people are using Brownie these days.

Anyway, thanks for the update @PatrickAlphaC !
